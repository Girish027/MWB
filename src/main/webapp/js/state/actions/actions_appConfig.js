import 'regenerator-runtime/runtime';

import getUrl, { pathKey } from 'utils/apiUrls';

import { fetchUtil } from 'utils/fetchUtils';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { initializeAnalytics, logAmplitudeEvent } from 'utils/amplitudeUtils';
import api from 'utils/api';
import * as types from './types';

const appConfigSuccess = data => ({
  type: types.APP_CONFIG_SUCCESS,
  data,
});

const appConfigFailure = data => ({
  type: types.APP_CONFIG_FAILURE,
  data,
});

const appConfigRequest = () => ({
  type: types.APP_CONFIG_REQUEST,
});

export const onAppConfigSuccess = data => (dispatch, getState) => {
  dispatch(appConfigSuccess(data));
  const state = getState();
  initializeAnalytics(state);
  logAmplitudeEvent(AmplitudeConstants.LOGIN_EVENT, state);
};

export const onAppConfigFailure = error => (dispatch) => {
  dispatch(appConfigFailure(error));
};

export const fetchAppConfig = () => (dispatch, getState) => {
  dispatch(appConfigRequest());
  const fetchUrl = getUrl(pathKey.appConfig);

  api.get({
    url: fetchUrl,
    onApiSuccess: onAppConfigSuccess,
    onApiError: onAppConfigFailure,
    dispatch,
    getState,
  });
};
