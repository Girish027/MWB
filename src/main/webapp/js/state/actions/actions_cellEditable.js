import * as types from './types';

export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_CREATE,
  stateKey,
});

export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_REMOVE,
  stateKey,
});

export const setActiveCell = ({ stateKey, activeRowIndex, activeColumnIndex }) => ({
  type: types.CELL_EDITABLE_SET_ACTIVE_CELL,
  stateKey,
  activeRowIndex,
  activeColumnIndex,
});

export const updateValue = ({ stateKey, value }) => ({
  type: types.CELL_EDITABLE_UPDATE_VALUE,
  stateKey,
  value,
});
