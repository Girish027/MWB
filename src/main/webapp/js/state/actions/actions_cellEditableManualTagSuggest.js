import store from 'state/configureStore';
import errorMessageUtil from 'utils/ErrorMessageUtil';
import getUrl, { pathKey } from 'utils/apiUrls';
import * as types from './types';

let controller;
export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_CREATE,
  stateKey,
});
export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_REMOVE,
  stateKey,
});

const getSuggestedTagsRequest = ({ stateKey, projectId, intent }) => ({
  type: types.GET_SUGGESTED_TAGS_REQUEST,
  stateKey,
  projectId,
  intent,
});

const getSuggestedTagsError = ({
  stateKey, projectId, intent, error,
}) => ({
  type: types.GET_SUGGESTED_TAGS_ERROR,
  stateKey,
  projectId,
  intent,
  error,
});

const getSuggestedTagsSuccess = ({
  stateKey, projectId, intent, suggestedTags,
}) => ({
  type: types.GET_SUGGESTED_TAGS_SUCCESS,
  stateKey,
  projectId,
  intent,
  suggestedTags,
});

export const getSuggestedTags = ({ stateKey, projectId, intent }) => (dispatch, getState) => new Promise((resolve, reject) => {
  if (controller) {
    controller.abort();
  }
  controller = new AbortController();
  const signal = controller.signal;
  dispatch(getSuggestedTagsRequest({ stateKey, projectId, intent }));
  const state = getState();
  const { csrfToken } = state.app;
  const clientId = state.header.client.id;

  const fetchUrl = getUrl(pathKey.suggestedTags, { projectId, intent, clientId });

  fetch(fetchUrl, {
    signal,
    credentials: 'same-origin',
    headers: {
      'X-CSRF-TOKEN': csrfToken,
    },
  })
    .then(errorMessageUtil.handleErrors)
    .then(response => response.json())
    .then((json) => {
      const suggestedTags = json && json.length ? json : [];
      const intentIndex = suggestedTags.indexOf(intent);
      if (intentIndex >= 0) {
        suggestedTags.slice(intentIndex, 1);
      }
      dispatch(getSuggestedTagsSuccess({
        stateKey, projectId, intent, suggestedTags,
      }));
      resolve([...suggestedTags]);
    })
    .catch((error) => {
      if (error.name === 'AbortError') { return; }
      dispatch(getSuggestedTagsError({
        stateKey, projectId, intent, error,
      }));
      reject(error);
    });
});

export const cursorDown = ({ stateKey }) => ({
  type: types.CELL_SUGGEST_CURSOR_DOWN,
  stateKey,
});

export const cursorUp = ({ stateKey }) => ({
  type: types.CELL_SUGGEST_CURSOR_UP,
  stateKey,
});
