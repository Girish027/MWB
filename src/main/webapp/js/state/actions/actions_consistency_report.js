import errorMessageUtil from 'utils/ErrorMessageUtil';
import store from 'state/configureStore';

import getUrl, { pathKey } from 'utils/apiUrls';
import * as appActions from './actions_app';
import * as types from './types';

export const reset = () => ({
  type: types.CONSISTENCY_REPORT_RESET,
});

export const setProjectId = projectId => ({
  type: types.CONSISTENCY_REPORT_SET_PROJECT_ID,
  projectId,
});

export const setProject = ({ projectId, project }) => ({
  type: types.CONSISTENCY_REPORT_SET_PROJECT,
  projectId,
  project,
});

export const setDatasets = ({ projectId, datasets }) => ({
  type: types.CONSISTENCY_REPORT_SET_DATASETS,
  projectId,
  datasets,
});


export const setFilter = ({ datasets, onlyConflicts }) => ({
  type: types.CONSISTENCY_REPORT_SET_FILTER,
  datasets,
  onlyConflicts,
});

const requestSearchSuccess = ({ results, startIndex }) => ({
  type: types.CONSISTENCY_REPORT_REQUEST_SEARCH_SUCCESS,
  results,
  startIndex,
});

const requestSearchRequest = ({
  promise, projectId, query, filter, sort, startIndex, limit,
}) => ({
  type: types.CONSISTENCY_REPORT_REQUEST_SEARCH_REQUEST,
  promise,
  projectId,
  query,
  filter,
  sort,
  startIndex,
  limit,
});

const requestSearchError = ({ error }) => ({
  type: types.CONSISTENCY_REPORT_REQUEST_SEARCH_ERROR,
  error,
});

export const setPagerSettings = ({ startIndex, limit, showControls }) => ({
  type: types.CONSISTENCY_REPORT_SET_PAGER_SETTINGS,
  startIndex,
  limit,
  showControls,
});

export const requestSearch = ({
  projectId, query, filter, sort, startIndex, limit, delay,
}, isNested) => (dispatch, getState) => {
  const promise = new Promise((resolve, reject) => {
    const state = getState();
    const { userId, csrfToken } = state.app;
    const clientId = state.header.client.id;

    const fetchUrl = getUrl(pathKey.consistency, {
      projectId,
      startIndex,
      limit,
      sortProperty: sort.property,
      sortDirection: sort.direction,
      clientId,
    });

    const filterBodyObj = {};
    filterBodyObj.onlyTagged = true;
    filterBodyObj.onlyConflicts = filter.onlyConflicts;
    if (filter.datasets && filter.datasets.length) {
      filterBodyObj.datasetIds = filter.datasets;
    }

    setTimeout(() => {
      fetch(fetchUrl, {
        credentials: 'same-origin',
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'X-CSRF-TOKEN': csrfToken,
        },
        body: JSON.stringify({
          configId: '0',
          filter: filterBodyObj,
        }),
      })
        .then(errorMessageUtil.handleErrors)
        .then(response => response.json())
        .then((results) => {
          const searchResults = results.transcriptionList || [];

          searchResults.forEach((res, index) => {
            res._rowIndex = index;
          });

          if (!searchResults.length && startIndex > 0) {
            const total = results.total;
            const maxPage = Math.max(Math.floor(total / limit) + (total % limit ? 1 : 0), 1);
            const newStartIndex = (maxPage - 1) * limit;
            dispatch(requestSearch({
              projectId,
              query,
              filter,
              sort,
              startIndex: newStartIndex,
              limit,
            }, true))
              .then((results) => {
                dispatch(requestSearchSuccess({ results, startIndex: newStartIndex }));
                resolve(results);
              })
              .catch((error) => {
                reject(error);
              });
          } else {
            if (!isNested) {
              dispatch(requestSearchSuccess({ results, startIndex }));
            }
            resolve(results);
          }
        })
        .catch((error) => {
          dispatch(requestSearchError({ error }));
          errorMessageUtil.dispatchError(error, dispatch, appActions.displayBadRequestMessage);
          reject(error);
        });
    }, delay || 0);
  });

  if (!isNested) {
    dispatch(requestSearchRequest({
      promise, projectId, query, filter, sort, startIndex, limit,
    }));
  }

  return promise;
};

export const refresh = ({ delay }) => ({
  type: types.CONSISTENCY_REPORT_REFRESH,
  delay,
});
