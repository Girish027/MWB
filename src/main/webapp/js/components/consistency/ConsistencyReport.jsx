import React, { Component } from 'react';
import { connect } from 'react-redux';
import Model from 'model';
import { Button, RadioGroup } from '@tfs/ui-components';
import * as actionsConsistencyReport from 'state/actions/actions_consistency_report';
import * as actionsCellEditable from 'state/actions/actions_cellEditable';
import * as actionscellEditableManualTagSuggest from 'state/actions/actions_cellEditableManualTagSuggest';
import { Actions as GridActions } from 'react-redux-grid';
import TaggerGridPager from 'components/controls/grid/TaggerGridPager';
import MultipleCheckboxFilter from 'components/controls/MultipleCheckboxFilter';
import getIcon, { IconNames } from 'utils/iconHelpers';
import { stateKey } from 'components/consistency/ConsistencyReportGrid';
import ConsistencyReportGrid from 'components/consistency/ConsistencyReportGrid';

export class ConsistencyReport extends Component {
  constructor(props, context) {
    super(props, context);

    this.controlsEl = null;
    this.cellEditableStateKey = 'ConsistencyReport';

    this.performSearch = this.performSearch.bind(this);
    this.onChangeRadioButton = this.onChangeRadioButton.bind(this);
    this.onChangeMultipleCheckboxFilter = this.onChangeMultipleCheckboxFilter.bind(this);
    this.onClickConsistencyReportRun = this.onClickConsistencyReportRun.bind(this);
    this.onChangeTaggerGridPager = this.onChangeTaggerGridPager.bind(this);

    this.completedDatasetsCount = 0;

    this.initialized = false;
    this.radios = {
      labels: ['Show All', 'Show Transcriptions With Different Suggested Tags'],
      values: [false, true],
    };
    this.props = props;
    this.state = {
      loadingProject: false,
      loadingDatasets: false,
      runPressed: false,
      selectedRadioGroupValue: props.consistencyReport.filter.onlyConflicts || false,
    };
  }

  componentWillReceiveProps(nextProps) {
    const { consistencyReport } = this.props;
    const { selectedProjectId, datasets } = nextProps;
    if (
      (consistencyReport.projectId && selectedProjectId && consistencyReport.projectId !== selectedProjectId)
      || (!consistencyReport.projectId && selectedProjectId)
      || (consistencyReport.projectId && !selectedProjectId)
    ) {
      this.props = nextProps;
      this.initiate();
    } else {
      if (consistencyReport.showControls != nextProps.consistencyReport.showControls) {
        setTimeout(() => { this.forceUpdate(); }, 100);
      }
      this.props = nextProps;
      if (this.props.consistencyReport.doSearch) {
        this.performSearch(this.props.consistencyReport.delay);
      }
    }

    if (datasets) {
      const { consistencyReport } = this.props;
      const { filter } = consistencyReport;
      const datasetsFilter = filter.datasets;
      let forceUpdate = false;
      const completedDatasetsIds = [];

      datasets.forEach((d) => {
        if (d.id && d.status && d.status == 'COMPLETED' && d.name) {
          completedDatasetsIds.push(d.id);
        }
      });

      if (completedDatasetsIds.length != this.completedDatasetsCount) {
        this.completedDatasetsCount = completedDatasetsIds.length;
        forceUpdate = true;
      }

      if (!datasetsFilter.length && completedDatasetsIds.length) {
        forceUpdate = false;
        setTimeout(() => {
          this.props.dispatch(actionsConsistencyReport.setFilter({
            datasets: completedDatasetsIds,
          }));
        }, 100);
      }

      if (forceUpdate) {
        setTimeout(() => {
          this.forceUpdate();
          if (!this.initialized) {
            // this.performSearch(this.props.consistencyReport.delay);
            this.setState({
              runPressed: true,
            });
            this.initialized = true;
          }
        }, 100);
      }
    }
  }

  componentDidMount() {
    this.completedDatasetsCount = 0;
    this.setState({
      loadingProject: false,
      loadingDatasets: false,
      runPressed: false,
    });
    const { dispatch } = this.props;
    dispatch(actionsCellEditable.stateCreate({ stateKey: this.cellEditableStateKey }));
    dispatch(actionscellEditableManualTagSuggest.stateCreate({ stateKey: this.cellEditableStateKey }));
    this.initiate();
  }

  initiate() {
    const {
      consistencyReport, datasets, selectedProjectId, dispatch,
    } = this.props;

    if (selectedProjectId) {
      if (selectedProjectId !== consistencyReport.projectId) {
        dispatch(actionsConsistencyReport.setProjectId(selectedProjectId));

        if (datasets && !consistencyReport.filter.datasets.length) {
          const datasetsIds = [];
          datasets.forEach((d) => {
            if (d.id && d.status && d.status == 'COMPLETED' && d.name) {
              datasetsIds.push(d.id);
            }
          });
          dispatch(actionsConsistencyReport.setFilter({
            datasets: datasetsIds,
          }));
        }
      }
    } else {
      dispatch(actionsConsistencyReport.setProjectId(null));
    }
  }

  componentWillUnmount() {
    const { dispatch } = this.props;
    dispatch(actionsConsistencyReport.reset());
    dispatch(GridActions.GridActions.setData({ data: [], stateKey }));
    dispatch(actionsCellEditable.stateRemove({ stateKey: this.cellEditableStateKey }));
    dispatch(actionscellEditableManualTagSuggest.stateRemove({ stateKey: this.cellEditableStateKey }));
  }

  get currentPage() {
    const { startIndex, limit } = this.props.consistencyReport;
    return Math.floor(startIndex / limit) + 1;
  }

  get maxPage() {
    const { total, limit } = this.props.consistencyReport;
    return Math.max(Math.floor(total / limit) + (total % limit ? 1 : 0), 1);
  }

  performSearch(delay) {
    const {
      projectId, query, filter, sort, startIndex, limit,
    } = this.props.consistencyReport;
    const { dispatch } = this.props;
    dispatch(actionsCellEditable.stateReset({ stateKey: this.cellEditableStateKey }));
    dispatch(actionscellEditableManualTagSuggest.stateReset({ stateKey: this.cellEditableStateKey }));
    dispatch(actionsConsistencyReport.requestSearch({
      projectId,
      query,
      filter,
      sort,
      startIndex,
      limit,
      delay,
    })).then(() => {
      dispatch(GridActions.GridActions.setData({ data: this.props.consistencyReport.searchResults, stateKey }));
    });
  }

  onClickConsistencyReportRun() {
    const { runPressed } = this.state;
    if (!runPressed) {
      this.setState({
        runPressed: true,
      });
    } else {
      this.performSearch();
    }
  }

  onChangeRadioButton(selected) {
    let selectedValueInBoolean = false;
    const { dispatch } = this.props;
    if (selected === 'false') {
      selectedValueInBoolean = false;
    } else if (selected === 'true') {
      selectedValueInBoolean = true;
    }
    this.setState({
      selectedRadioGroupValue: selectedValueInBoolean,
    });
    dispatch(actionsConsistencyReport.setFilter({ onlyConflicts: selectedValueInBoolean }));
  }

  onChangeMultipleCheckboxFilter(newValue) {
    const { dispatch } = this.props;
    dispatch(actionsConsistencyReport.setFilter({ datasets: newValue }));
  }

  onChangeTaggerGridPager(startIndex, limit, showControls) {
    const { dispatch } = this.props;
    dispatch(actionsConsistencyReport.setPagerSettings({ startIndex, limit, showControls }));
  }

  render() {
    const {
      consistencyReport, project, datasets,
    } = this.props;
    const {
      filter, isSearching, isError,
      showControls, limit, startIndex,
      total, isUpdatingTags,
    } = consistencyReport;
    const { loadingProject, loadingDatasets, runPressed } = this.state;

    if (loadingProject || loadingDatasets) {
      return (<div id="ConsistencyReport" />);
    }

    if (!project) {
      return (
        <div id="ConsistencyReport">
          <p>Requested project is not found.</p>
        </div>
      );
    }

    if (!datasets || !this.completedDatasetsCount) {
      return (
        <div id="ConsistencyReport">
          <p>Requested project does not have transformed datasets.</p>
        </div>
      );
    }

    let gridPaddingTop = 0;
    if (showControls && this.controlsEl && this.controlsEl.boxContainerRef && this.controlsEl.boxContainerRef.offsetHeight) {
      gridPaddingTop += this.controlsEl.boxContainerRef.offsetHeight;
    }

    const gridMessage = isSearching || isUpdatingTags ? 'Loading...' : (isError ? 'There was error processing your request.' : '');

    const datasetsOptions = [];
    datasets.forEach((d) => {
      if (d.id && d.status && d.status == 'COMPLETED' && d.name) {
        datasetsOptions.push({ value: d.id, label: d.name });
      }
    });

    return (
      <div id="ConsistencyReport">
        {showControls ? (
          <div
            id="ConsistencyReportControls"
            className="TaggerGridControls"
            direction="column"
            ref={(div) => { this.controlsEl = div; }}
          >

            <MultipleCheckboxFilter
              label="Datasets Filter:"
              className="DatasetsFilter"
              options={datasetsOptions}
              value={filter.datasets}
              onChange={this.onChangeMultipleCheckboxFilter}
              allowEmpty={false}
            />
            <div className="searchContainer" direction="row">
              <div
                className="filters"
                direction="column"
              >
                <div className="filtersBlock">
                  <RadioGroup
                    displayType="stacked"
                    onChange={this.onChangeRadioButton}
                    values={this.radios.values}
                    labels={this.radios.labels}
                    value={this.state.selectedRadioGroupValue}
                  />
                </div>
              </div>
              <div className="actions" direction="column">
                <Button
                  name="consistency-run"
                  id="ConsistencyReportRun"
                  onClick={isSearching || isUpdatingTags ? null : this.onClickConsistencyReportRun}
                  icon={getIcon(IconNames.CONSISTENCY)}
                  data-qa="consistency-run"
                >
                  RUN
                </Button>
              </div>
            </div>
          </div>
        ) : null}

        {isSearching || isUpdatingTags || isError ? (
          <div className="GridMessage" style={{ paddingTop: gridPaddingTop }}>
            {gridMessage}
          </div>
        ) : null}

        {runPressed && !isUpdatingTags ? (
          <ConsistencyReportGrid
            paddingTop={gridPaddingTop}
            maxPage={this.maxPage}
            currentPage={this.currentPage}
            total={total}
            onTagsChange={() => { this.performSearch(3000); }}
          />
        ) : null}

        {!isSearching && !isUpdatingTags ? (
          <TaggerGridPager
            onChange={({ startIndex, limit, showControls }) => this.onChangeTaggerGridPager(startIndex, limit, showControls)}
            total={total}
            limit={limit}
            startIndex={startIndex}
            showControls={showControls}
            controls={
              TaggerGridPager.CONTROLS.TOGGLE_CONTROLS
              + (runPressed ? TaggerGridPager.CONTROLS.RESULTS_PER_PAGE : 0)
              + (runPressed ? TaggerGridPager.CONTROLS.CHOOSE_EXACT_PAGE : 0)
              + (runPressed && this.maxPage > 2 ? TaggerGridPager.CONTROLS.PAGE_BUTTONS : 0)
            }
          />
        ) : null}

      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  consistencyReport: state.consistencyReport,
  projectsManager: state.projectsManager, // TODO: improvements
  selectedProjectId: state.projectListSidebar.selectedProjectId,
  project: Model.ProjectsManager.getProject(state.projectListSidebar.selectedProjectId) || null,
  datasets: Model.ProjectsManager.getDatasetsByProjectId(state.projectListSidebar.selectedProjectId, true) || null,
});

const mapDispatchToProps = dispatch => ({
  dispatch,
});

export default connect(mapStateToProps, mapDispatchToProps)(ConsistencyReport);
