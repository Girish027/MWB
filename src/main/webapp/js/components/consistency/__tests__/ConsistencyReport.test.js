import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter, Switch, Route } from 'react-router';
import store from 'state/configureStore';
import ReduxConsistencyReport from 'components/consistency/ConsistencyReport';
import { ConsistencyReport } from 'components/consistency/ConsistencyReport';
import toJSON from 'enzyme-to-json';
import * as actionsConsistencyReport from 'state/actions/actions_consistency_report';
import * as actionsCellEditable from 'state/actions/actions_cellEditable';
import * as actionscellEditableManualTagSuggest from 'state/actions/actions_cellEditableManualTagSuggest';

describe('<ConsistencyReport />', () => {
  let wrapper;
  const datasets = [{ id: '2', status: 'COMPLETED', name: 'abc' },
    { id: '3', status: 'COMPLETED', name: 'abc1' }];
  const consistencyReport = {
    filter: { datasets: [] },
    projectId: '12',
    query: 'sd',
    sort: 'ASC',
    isSearching: false,
    isError: false,
    showControls: true,
    limit: 10,
    startIndex: 0,
    total: 20,
    isUpdatingTags: false,
  };
  const props = {
    consistencyReport,
    selectedProjectId: '13',
    datasets,
    dispatch: jest.fn(),
    project: {
      id: '12',
      name: 'project1',
    },
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (propsObj) => shallow(<ConsistencyReport
    {...propsObj}
  />);

  describe('Creating an instance', () => {
    beforeAll(() => {
      wrapper = mount(<Provider store={store}>
        <MemoryRouter
          initialEntries={['/resolve-inconsistency']}
          initialIndex={0}
        >
          <Switch>
            <Route path="/resolve-inconsistency" component={ReduxConsistencyReport} />
          </Switch>
        </MemoryRouter>
      </Provider>);

      wrapper.setProps({
        selectedProjectId: '-1',
        datasets: new Map(),
        project: {},
      });
    });

    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });

    test('should contain ConsistencyReport', () => {
      expect(wrapper.find(ReduxConsistencyReport).length).toBe(1);
    });

    test('should not contain div#ConsistencyReportGridContainer', () => {
      const find = wrapper.find('div#ConsistencyReportGridContainer');
      expect(find).toHaveLength(0);
    });
  });

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = getShallowWrapper(props);
      expect(wrapper.exists()).toBe(true);
    });

    test('should exist with state set', () => {
      wrapper = getShallowWrapper(props);
      wrapper.setState({ loadingProject: false, loadingDatasets: true, runPressed: true });
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly for default props on mount', () => {
      wrapper = mount(<ConsistencyReport
        {...props}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly for default props on shallow copy', () => {
      wrapper = getShallowWrapper(props);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    beforeEach(() => {
      props.dispatch = jest.fn();
      actionsCellEditable.stateReset = jest.fn(() => 'called actionsCellEditable stateReset');
      actionscellEditableManualTagSuggest.stateReset = jest.fn(() => 'called actionscellEditableManualTagSuggest stateReset');
      actionsConsistencyReport.requestSearch = jest.fn(() => 'called requestSearch');
      actionsConsistencyReport.setFilter = jest.fn(() => 'called setFilter');
      actionsConsistencyReport.setPagerSettings = jest.fn(() => 'called setPagerSettings');
      actionsCellEditable.stateRemove = jest.fn(() => 'called actionsCellEditable stateRemove');
      actionsConsistencyReport.reset = jest.fn(() => 'called reset');
      actionscellEditableManualTagSuggest.stateRemove = jest.fn(() => 'called actionscellEditableManualTagSuggest stateRemove');
    });

    describe('onClickConsistencyReportRun', () => {
      test('should call onClickConsistencyReportRun with runPressed as true', () => {
        wrapper = getShallowWrapper(props);
        wrapper.setState({ runPressed: true });
        wrapper.instance().performSearch = jest.fn();
        wrapper.instance().onClickConsistencyReportRun();
        expect(wrapper.instance().performSearch).toHaveBeenCalledWith();
      });

      test('should call onClickConsistencyReportRun with runPressed as false', () => {
        wrapper = getShallowWrapper(props);
        wrapper.setState({ runPressed: false });
        wrapper.instance().onClickConsistencyReportRun();
        expect(wrapper.state().runPressed).toBe(true);
      });
    });

    describe('onChangeRadioButton', () => {
      test('should call onChangeRadioButton with selected', () => {
        const selected = 'false';
        wrapper = getShallowWrapper(props);
        wrapper.instance().onChangeRadioButton(selected);
        expect(actionsConsistencyReport.setFilter).toHaveBeenCalledWith({ onlyConflicts: false });
        expect(props.dispatch).toHaveBeenCalledWith('called setFilter');
      });
    });

    describe('onChangeMultipleCheckboxFilter', () => {
      test('should call onChangeMultipleCheckboxFilter with selected', () => {
        const newValue = '123';
        wrapper = getShallowWrapper(props);
        wrapper.instance().onChangeMultipleCheckboxFilter(newValue);
        expect(actionsConsistencyReport.setFilter).toHaveBeenCalledWith({ datasets: newValue });
        expect(props.dispatch).toHaveBeenCalledWith('called setFilter');
      });
    });

    describe('onChangeMultipleCheckboxFilter', () => {
      test('should call onChangeMultipleCheckboxFilter with limit, startIndex, and showControls', () => {
        const limit = 10;
        const startIndex = 0;
        const showControls = true;
        wrapper = getShallowWrapper(props);
        wrapper.instance().onChangeTaggerGridPager(startIndex, limit, showControls);
        expect(actionsConsistencyReport.setPagerSettings).toHaveBeenCalledWith({ startIndex, limit, showControls });
        expect(props.dispatch).toHaveBeenCalledWith('called setPagerSettings');
      });
    });

    describe('componentWillUnmount', () => {
      test('should call componentWillUnmount', () => {
        wrapper = getShallowWrapper(props);
        wrapper.instance().cellEditableStateKey = 'stateKey';
        wrapper.instance().componentWillUnmount();
        expect(actionsConsistencyReport.reset).toHaveBeenCalledWith();
        expect(actionsCellEditable.stateRemove).toHaveBeenCalledWith({ stateKey: wrapper.instance().cellEditableStateKey });
        expect(actionscellEditableManualTagSuggest.stateRemove).toHaveBeenCalledWith({ stateKey: wrapper.instance().cellEditableStateKey });
        expect(props.dispatch).toHaveBeenCalledWith('called actionsCellEditable stateRemove');
        expect(props.dispatch).toHaveBeenCalledWith('called reset');
        expect(props.dispatch).toHaveBeenCalledWith('called actionscellEditableManualTagSuggest stateRemove');
      });
    });

    describe('componentWillReceiveProps', () => {
      test('should call componentWillReceiveProps with nextProps', () => {
        const nextProps = {
          ...props,
        };
        wrapper = getShallowWrapper(props);
        wrapper.instance().componentWillReceiveProps(nextProps);
        expect(props.dispatch).toHaveBeenCalledWith({ stateKey: 'ConsistencyReport', type: 'CELL_EDITABLE_STATE_CREATE' });
      });
    });
  });
});
