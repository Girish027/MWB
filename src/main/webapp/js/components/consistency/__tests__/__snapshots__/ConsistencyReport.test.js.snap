// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConsistencyReport /> Snapshots renders correctly for default props on mount 1`] = `
<ConsistencyReport
  consistencyReport={
    Object {
      "filter": Object {
        "datasets": Array [],
      },
      "isError": false,
      "isSearching": false,
      "isUpdatingTags": false,
      "limit": 10,
      "projectId": "12",
      "query": "sd",
      "showControls": true,
      "sort": "ASC",
      "startIndex": 0,
      "total": 20,
    }
  }
  datasets={
    Array [
      Object {
        "id": "2",
        "name": "abc",
        "status": "COMPLETED",
      },
      Object {
        "id": "3",
        "name": "abc1",
        "status": "COMPLETED",
      },
    ]
  }
  dispatch={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "stateKey": "ConsistencyReport",
            "type": "CELL_EDITABLE_STATE_CREATE",
          },
        ],
        Array [
          Object {
            "stateKey": "ConsistencyReport",
            "type": "CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_CREATE",
          },
        ],
        Array [
          Object {
            "projectId": "13",
            "type": "CONSISTENCY_REPORT_SET_PROJECT_ID",
          },
        ],
        Array [
          Object {
            "datasets": Array [
              "2",
              "3",
            ],
            "onlyConflicts": undefined,
            "type": "CONSISTENCY_REPORT_SET_FILTER",
          },
        ],
        Array [
          Object {
            "stateKey": "ConsistencyReport",
            "type": "CELL_EDITABLE_STATE_CREATE",
          },
        ],
        Array [
          Object {
            "stateKey": "ConsistencyReport",
            "type": "CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_CREATE",
          },
        ],
        Array [
          Object {
            "projectId": "13",
            "type": "CONSISTENCY_REPORT_SET_PROJECT_ID",
          },
        ],
        Array [
          Object {
            "datasets": Array [
              "2",
              "3",
            ],
            "onlyConflicts": undefined,
            "type": "CONSISTENCY_REPORT_SET_FILTER",
          },
        ],
        Array [
          Object {
            "stateKey": "ConsistencyReport",
            "type": "CELL_EDITABLE_STATE_CREATE",
          },
        ],
        Array [
          Object {
            "stateKey": "ConsistencyReport",
            "type": "CELL_EDITABLE_MANUAL_TAG_SUGGEST_STATE_CREATE",
          },
        ],
        Array [
          Object {
            "projectId": "13",
            "type": "CONSISTENCY_REPORT_SET_PROJECT_ID",
          },
        ],
        Array [
          Object {
            "datasets": Array [
              "2",
              "3",
            ],
            "onlyConflicts": undefined,
            "type": "CONSISTENCY_REPORT_SET_FILTER",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  project={
    Object {
      "id": "12",
      "name": "project1",
    }
  }
  selectedProjectId="13"
>
  <div
    id="ConsistencyReport"
  >
    <p>
      Requested project does not have transformed datasets.
    </p>
  </div>
</ConsistencyReport>
`;

exports[`<ConsistencyReport /> Snapshots renders correctly for default props on shallow copy 1`] = `
<div
  id="ConsistencyReport"
>
  <p>
    Requested project does not have transformed datasets.
  </p>
</div>
`;
